doc/decisions/validation.md

```diff
@@ -0,0 +1,63 @@
+# Validation
+
+## Issue
+
+Validation plugins operate as indenpendent blackboxes. 
+For every backend each mounted validation plugin iterates 
+over the whole keyset, checks every key for it's trigger metakey, 
+and validates the key.
+
+Each plugin decides on it's own what should happen when a key 
+fails to validate. 
+
+## Constraints
+
+## Assumptions
+
+While plugins should always fail and return
+an error if validation fails on kdbSet, there are be different
+requirements on what should happen on kdbGet:
```

Comment:
some rationale why we would need this. I do not think all of these usecases actually are useful.